src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -1009,6 +1013,57 @@ private static void ReportUseSiteDiagnostics<TMember>(ArrayBuilder<MemberResolut
             }
         }
 
+        private int GetTheBestCandidateIndex<TMember>(ArrayBuilder<MemberResolutionResult<TMember>> results, AnalyzedArguments arguments, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
+            where TMember : Symbol
+        {
+            int currentBestIndex = -1;
+            for (int index = 0; index < results.Count; index++)
+            {
+                if (!results[index].IsValid)
+                {
+                    continue;
+                }
+
+                // Assume that the current candidate is the best if we don't have any
+                if (currentBestIndex == -1)
+                {
+                    currentBestIndex = index;
+                }
+                else if (results[currentBestIndex].Member != results[index].Member)
```

Comment:
>else if (results[currentBestIndex].Member != results[index].Member) [](start = 16, length = 67)

It feels like currentBestIndex should be reset to -1 if this condition is false. Imagine what would be the result of the old implementation and the new implementation if we have only to candidates in the set and both are based on the same member.